File: src/main/kotlin/rpa/mapper/ScheduleToDTOMapper.java
package rpa.mapper;

import org.springframework.stereotype.Component;
import rpa.model.*;

import java.util.List;
import java.util.stream.Collectors;

@Component
public class ScheduleToDTOMapper {

    public EventResponseDTO map(EventEntity eventEntity) {
        return new EventResponseDTO(eventEntity.getStartTime(), eventEntity.getType(), eventEntity.getLeagueName(), mapMatch(eventEntity.getMatch()));
    }

    private MatchResponseDTO mapMatch(MatchEntity matchEntity) {
        return new MatchResponseDTO(matchEntity.getMatchId(), matchEntity.getGameCount(), mapTeams(matchEntity.getTeams()));
    }

    private List<TeamsResponseDTO> mapTeams(List<TeamEntity> teamEntities) {
        return teamEntities.stream()
                .map(t -> new TeamsResponseDTO(t.getCode(), t.getName(), t.getImage(), t.getRecordWins() != null ? t.getRecordWins() : 0, t.getRecordLosses() != null ? t.getRecordLosses() : 0))
                .collect(Collectors.toList());
    }
}

File: src/main/kotlin/rpa/mapper/ScheduleToDTOMapper.kt
//package rpa.mapper
//
//import rpa.model.*
//import java.util.stream.Collectors
//
//public class ScheduleToDTOMapper {
//
//    fun map(eventEntity : EventEntity) : EventResponseDTO {
//        return EventResponseDTO(
//                startTime = eventEntity.startTime,
//                type = eventEntity.type,
//                leagueName = eventEntity.leagueName,
//                match = mapMatch(eventEntity.match)
//
//                )
//    }
//
//    private fun mapMatch(matchEntity : MatchEntity) : MatchResponseDTO {
//        return MatchResponseDTO(
//                matchId = matchEntity.matchId,
//                gameCount = matchEntity.gameCount,
//                teams = mapTeams(matchEntity.teams)
//        )
//    }
//
//    private fun mapTeams(teamEntities : List<TeamEntity>) : List<TeamsResponseDTO> {
//        return teamEntities.stream().map {
//            t: TeamEntity -> TeamsResponseDTO(
//                code = t.code,
//                name = t.name,
//                image = t.image,
//                recordLosses = t.recordLosses,
//                recordWins = t.recordWins
//        )
//        }.collect(Collectors.toList())
//    }
//
//
//}